### PR TITLE
Tritium now produces water vapor when combusting.

### PR DESCRIPTION
**Pull request by Buggy123**
Tritum is a fucking isotope of hydrogen it does not produce **CARBON DIOXIDE** when burning. H+O does not equal CO2. Technically it makes Tritiated Water but I'm not adding a radioactive form of water vapor unless a maintainer wants it so plain water it is.

Also we need a way to produce water vapor anyway so here you go.

🆑 
tweak: After consulting with their in-house physicists, Nanotrasen has updated their worst-case disaster training simulation "Space Station 13". The combustion of hydrogen isotopes now produces water vapor instead of carbon dioxide.
/🆑

Might conflict a little with #35348 as it adds a define used by it. Could eventually be expanded so that tritium mimics the behavior of hydrogen if desired.

### CODE REVIEW

**Comment by arsserpentarium** [comment]
You should make burning of tritium water radioactive.If there is no tritium water,let's count,that it releases all the radiation while burning.

**Comment by Buggy123** [comment]
Sounds good to me, if anyone else is up for it I'll try and do it.

**Comment by DaedalusGame** [comment]
you have to change the combustion energy define too

**Comment by DaedalusGame** [comment]
Fusion does this

**Comment by Buggy123** [comment]
Yeah I think it'd be fairly simply to copy/paste, I just want to know if anyone else wants that or not.

**Comment by nicbn** [comment]
Should result in tritiated water not water

**Comment by DaedalusGame** [comment]
Making Tritium fires radioactive atleast gives some merit in using it for power production/sm fuel

**Comment by Buggy123** [comment]
>Should result in tritium oxide not water. (Super heavy water is a cooler name)

I don't think it's a good idea to add another gas which behaves extremely similarly to a pre-existing one except for minor differences (bloat), so I'm not going to do it unless a maintainer okays it.

**Comment by ExcessiveUseOfCobblestone** [comment]
so the gas made by burning small amounts of plasma with lots of oxygen makes something that puts out further production when it's burnt?

**Comment by ExcessiveUseOfCobblestone** [comment]
Also you can produce water vapor it is just quite extensive (which is fair for aoe slip gas)

**Comment by DaedalusGame** [comment]
You can not produce water vapor.

**Comment by ExcessiveUseOfCobblestone** [comment]
You can, the SM makes it when exposed to Water Vapor.

**Comment by Buggy123** [comment]
Yeah, #35348 enables water production but only with hydrogen, and no one is sure yet how you'll get that.

Water vapor... doesn't... seem to put out fires? I can't seem to find anything to that effect. Certainly, it doesn't inherently put out fires in real life, not anymore than any other mostly-inert gas. _Liquid_ water will put out fires, but that's because its cold, takes lots of energy to heat up, and can flow to surround and cover the base of fires. Sand also works pretty well.

**Comment by DaedalusGame** [comment]
Idk if people just intentionally make shit up or what but if you can show me some actual code that declares that the SM dupes water vapor

**Comment by ExcessiveUseOfCobblestone** [comment]
#20236 to put out fire

I was wrong because it was Freon that did this as per #24785

**Comment by Darkgenerallord** [comment]
This will 100% extinguish the tritium fire.

**Comment by DaedalusGame** [comment]
Uh we just agreed that water vapor doesn't put out fires

**Comment by ExcessiveUseOfCobblestone** [comment]
No we agreed WV doesn't dupe on SM.

I linked the PR which puts out fires.

**Comment by DaedalusGame** [comment]
https://github.com/tgstation/tgstation/blob/master/code/game/turfs/open.dm#L170-L180

**Comment by JJRcop** [comment]
I think you should shorten the changelog entry

**Comment by Buggy123** [comment]
Oh fine, it's shorter. RIP fun.

**Comment by nicbn** [comment]
I think you shouldn't shorten the entry

**Comment by Buggy123** [comment]
Done.

**Comment by Buggy123** [comment]
Okay, burning Trit now produces radiation. Really no idea how much, could be absolutely absurd, could be nothing. Gonna have to test it a bit.

File: code/modules/atmospherics/gasmixtures/reactions.dm
Comment (DaedalusGame) [inline code/modules/atmospherics/gasmixtures/reactions.dm]: if (burned_fuel>0 && istype(location))

Comment (Buggy123) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: whoops

File: code/modules/atmospherics/gasmixtures/reactions.dm
Comment (DaedalusGame) [inline code/modules/atmospherics/gasmixtures/reactions.dm]: Why did this move down here aaaah

File: code/modules/atmospherics/gasmixtures/reactions.dm
Comment (DaedalusGame) [inline code/modules/atmospherics/gasmixtures/reactions.dm]: ok, let's assume we burn 1 mole of tritium in very oxygen rich enviroment.
tritium burns at 10x its molar mass in oxygen, so that's 10 moles worth of burn
burning one mole produces FIRE_HYDROGEN_ENERGY_RELEASED energy (280000)
burning 10 moles thus produces 2.8 million whatevers
the energy released divided by 1000 returns the strength of the rad pulse

tl;dr one mole of tritium produces 2800 rads

Comment (Buggy123) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: Okay, I sorta eyeballed it but I might've eyeballed it a bit too high. Let me test ingame and see how it goes.
But I asked coderbus, and they said it should have a lot of rads compared to the SM so...

**Comment by Buggy123** [comment]
image
Hmm...

**Comment by DaedalusGame** [comment]
hmmmmmmmm

**Comment by Buggy123** [comment]
I tried reviving myself but I just instantly die again because of how much radiation there is.

**Comment by arsserpentarium** [comment]
Best choice for flamethrower

**Comment by Buggy123** [comment]
I was wearing a radsuit. With 100% rad protection.

**Comment by duncathan** [comment]
>I don't think it's a good idea to add another gas which behaves extremely similarly to a pre-existing one except for minor differences (bloat), so I'm not going to do it unless a maintainer okays it.

You think correctly.

**Comment by duncathan** [comment]
@as334 I would like your thoughts on this

**Comment by Buggy123** [comment]
Okay it no longer spews enough radiation to sterilize the entire station, but still enough that I think you could concievably make power with it.

File: code/modules/atmospherics/gasmixtures/reactions.dm
Comment (vuonojenmustaturska) [inline code/modules/atmospherics/gasmixtures/reactions.dm]: I don't mind merging the part where tritium burns into water, but the rad memes are a bit much considering the relatively inert nature of tritium and `radiation_pulse()` not being free performance-wise.

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

**Comment by evsey9** [comment]
uh wouldnt tritium making water vapor cause tritium fires to be short lasted?

**Comment by DaedalusGame** [comment]
DGL says it does. But I can't believe that. This change makes every tritium mole burn 2.8x hotter, water vapor doesn't inately extinguish fires, albeit it does have a higher heat capacity. But also, only 1/100th of a water vapor mole is produced per mole of tritium burnt.

Someone should probably actually test if tritium breeders still work with this change. I can't really believe that 1.33x heat capacity is going to counteract 2.8x heat energy produced.

**Comment by Buggy123** [comment]
Also, tritium fires are fucking weird as hell. If there's more oxygen than tritium, it burns FAST and **HOT**, in practice a fire in the mixing chamber, consisting of 4 canisters of O2 and 1 of tritium, burned for about 5-10 seconds total and reached several hundred thousand degrees, enough to burn through the reinforced (nigh-fireproof) plasma glass window.

Once there's more tritium than oxygen, it then burns ENDOTHERMICALLY, _COOLING DOWN_ the air around it. With identical experiments (which used a rwall instead of a window so it didn't vent to space) it "burned" down to a temperature of only a few thousand degrees after 30-60 seconds or so.

None of that is because of me or this PR, for the record. Evidentally that is all intentional, and the only thing I did was make it burn slightly hotter. I also added a radiation pulse to it, 99% of while is in a intense burst during the initial burning phase and essentially nothing afterwards.

**Comment by ExcessiveUseOfCobblestone** [comment]
So the radiation pulse also makes plux r-r-r-right?

**Comment by arsserpentarium** [comment]
endothermic fires is totally bug, but it's that kind of bug, that we calls feature.don't remove it!

**Comment by DaedalusGame** [comment]
It's not a bug, it's supposed to die out if there's little oxygen.

File: code/modules/atmospherics/gasmixtures/reactions.dm
Comment (Buggy123) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: But... fusion does (did) the exact same thing, and it was considered fine (it was disabled for other reasons). Not only that, but tritum burns REALLY quickly so you looking at a couple seconds of this at most.

It might still pulse when it's doing the weird endothermic burn thing, so if you'd like I could change it so it only radiates during the brief, O2-rich burn.

**Comment by as334** [comment]
Hey, original person who made it do carbon dioxide not water vapor. This was 100% just a balance thing, I'm aware of how combustion works. It's just in testing I found that fires that slip you were pretty god damn insane so I changed it. But if people want it to be water I'm in favor.

**Comment by DaedalusGame** [comment]
Water Vapor should probably be changed to not produce wet floors past the evaporation point

**Comment by Buggy123** [comment]
To be fair, tritium fires are PRETTY rare and won't last long because they'll burn through the floor in all of 5 seconds. Also, I never see anyone use them as opposed to plasmafires, because tritium is much harder to obtain, so I don't think a buff is unreasonable.

File: code/modules/atmospherics/gasmixtures/reactions.dm
Comment (Buggy123) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: @vuonojenmustaturska

**Comment by Buggy123** [comment]
I'm waiting on @vuonojenmustaturska to reply as to whether he agrees with what I suggested as a compromise, or still wants me to remove the rad pulsing. If he insists I'd be alright with removing it, but I want his feedback here.

**Comment by arsserpentarium** [comment]
No,there is no point to burn tritium without radiation. Why even to do this?

**Pull request merged**
